torch/distributed/algorithms/model_averaging/averagers.py

```diff
@@ -101,13 +101,28 @@ def __init__(
 
     def average_parameters(self, params):
         r"""
-        Averages parameters if ``step`` is no less than ``warmup_steps``
+        Averages parameters or parameter groups of an optimizer if ``step`` is no less than ``warmup_steps``
         and it can be divided by ``period``, where ``step`` is increased by 1
         at each iteration in the training loop.
+        params: average model.parameters() or parameter groups of an optimizer
         """
         if (
             self.step >= self.warmup_steps
             and (self.step - self.warmup_steps) % self.period == 0
         ):
-            utils.average_parameters(iter(params), self.process_group)
+            filter_params = []
+            for param in params:
+                if isinstance(param, torch.nn.Parameter):
+                    # model.parameters() input
+                    param_data = param
+                    if param_data.grad is not None:
+                        filter_params.append(param_data)
+                elif isinstance(param, dict):
+                    # optimizer.param_groups input
+                    for param_data in param["params"]:
+                        if param_data.grad is not None:
+                            filter_params.append(param_data)
+                else:
+                    raise NotImplementedError
```

Comment:
`raise NotImplementedError(f"Parameter input of type {type(param)} is not supported")`

Reply:
fix